### ISSUE TITLE
Intel Manager/Survey123: Smart Editor not accepting "data source" as a Preset

### ISSUE THREAD

**Issue by pjdohertygis** [pjdohertygis/WideAreaSearchTemplates#144]
This is a known issue (Esri Web AppBuilder and Survey123 Team discussed) where fields created as hidden in Survey123 Connect do not quite work with Smart Editor Presets.

**Comment by avezina** [comment]
In the future the recommendation is to set the appearance to hidden rather than changing the type.
![image1](https://user-images.githubusercontent.com/50338557/123181224-bcc63080-d441-11eb-92ba-6c9b60e80d8d.png)

There is a fix if you modify the JSON.
The hidden template is missing from the list of templates and needs to be added back in.

Example:

Service with hidden field: 
https://services2.arcgis.com/PWJUSsdoJDp7SgLj/ArcGIS/rest/services/service_a5ea438834164403954a3ad6f4c4db10/FeatureServer/0?f=pjson&token=gqwQffAvKKW0Thx8K_lDRfmU_eAZ_tdkPy0HZJd4Bh6f49KgQo5fX9Zhvr08ztyDqqov6x6fcOBvgaXLS6ixUFVc5wfR6PMmbw7fkxw_G0pvYP148qx-GdQtOnvEQWl6TUx6R77SwFlxHuG52Q7cFQAM0ks9NuVELUKnMhjaeS3ePj4E_SMiSyaU_T7By_yXjwPQ38EQv1LCwobuYFeeStQmtA6lZGsJLLxCmJpFfeCiRjHC1zNbVvqYQiWOzkvT
![image](https://user-images.githubusercontent.com/50338557/123181442-26463f00-d442-11eb-8305-e1af41b584b1.png)

Service without hidden field:
https://services2.arcgis.com/PWJUSsdoJDp7SgLj/ArcGIS/rest/services/Form_123-republish/FeatureServer/0?f=pjson&token=dP7D3rVmqP6inHDhE0aH0GfqXnUONMK3iu0mriFVsF-SsbLZNHR9lI7IQp47K8d98jQlM8tfDxpet0Y4qkmZefwHUWh-hvHiIXlUVDlcHZtItfXiJ_g9Va_FzWhQ31C5VARfyA-YLcVaDXWLckC4SUuESUL64K1uZ2SrsBUjoILWTD8tXNVDU2dh2u3OBAtsRpXlFtdH-LPHWFd_IGiNWU-xerpDCWRF3996hvUegDndqRKqrAZC-GaHO8HyqGKS
![image](https://user-images.githubusercontent.com/50338557/123181477-3eb65980-d442-11eb-99b8-a7acd9c0d554.png)

**Comment by avezina** [comment]
@CLivingstone @pjdohertygis 
I don't know if you've fixed this issue in the JSON on your end, but I found another workaround that won't require updating the JSON. I don't know why I did not think of this earlier, I'm sorry!

So if something like this happens again, and you notice the field is not listed in the attributes as shown further above (it can happen for example if you publish a service with ArcGIS Pro without having created any editing templates for your features). Then, by managing and updating the editing templates in the web map, this will update the template and write it into the layer's JSON.
https://doc.arcgis.com/en/arcgis-online/manage-data/manage-feature-templates.htm#ESRI_SECTION1_AF3434562C3442F393C942E8952A3673

At this time, if a field is missing in the templates section of the JSON this may affect your Preset Attribute Actions and the Copy Features functionality. Honestly should not happen very often, but now you know if you happen to hit this issue again. We're working on a fix for the copy functionality.
